### ISSUE TITLE
Usage Question - best-practice for padding (zerofill) and trimming a BitArray?

### ISSUE THREAD

**Issue by obarak** [scott-griffiths/bitstring#228]
Usage question:
#### TL;DR:  ####
1. What's the proper way to left-pad a `BitArray` to a specified length? (like Python's `zfill()`)
2. What's the proper way to trim leading zeros from a `BitArray`? (like Python's `lstrip()`)

#### Details: ####
I'm reading inputs which are 16-bits, in some custom packing (e.g. bits 0-5 are "height" and bits 6-15 are "width").
I've defined an unpack format, but problem is that leading zeros in the inputs are trimmed, so I need to pad it back to 16 bits for the unpacking to work. Also, the inputs have '0b' prefix but are read in as strings and not literals.

So what I'm doing today is:

```python
input = '0b01110'
input = input.replace('0b','').zfill(16)
BitArray(bin=input).unpack(fmt)
```

I could do:
```python
input = '0b01110'
input = int(input, 2)
BitArray(uint=input, length=16).unpack(fmt)  # length param only works for int/uint inputs
```

But that still an extra step. I was wondering if there's a way I'm missing, something like:
```python
input = '0b01110'
BitArray(input).zfill(16).unpack(fmt) # not working as no zfill() method
```

or 
```python
input = '0b01110'
BitArray(input, length=16) # not working as length can't be used on non-ints
```

For trimming, there's a related use-case where I'm starting from a BitArray which is zero-padded to 8-bit, but is known to always contain a value of up to 5 bits. In this case I need to left-trim zeros from the BitArray up to 5 bits (but not more). 
So I'm doing:

```python
a=BitArray('0b00000001') # this is my 8-bit input
a_str = a.bin # get it as a python string
a_str = a_str.lstrip('0').zfill(5)
BitArray(bin=a_str) 
>> BitArray('0b00001') # desired output 
```

**Comment by scott-griffiths** [comment]
Hi.

I deliberately didn't add fill-like methods to the library as there isn't one obvious way of doing the filling, and a format string for specifying it was too easy to forget even for me. Some people might want to fill on the right, others the left, some with zeros (OK, most) and some with ones. So it ends up being a bit manual.

I'd either use a format that has a length, such as uint, and use that:

`b = BitArray(a.uint, length=16)`

or be more explicit:

`b = BitArray(16 - len(a)) + a   # Initialising with an integer gives a zeroed bitstring of that length`

For your second example I think you can always just trim the first 3 bits from the start - you don't need `lstrip` or `zfill`:

```
>>> a = BitArray('0b00000001')
>>> a[-5:]
BitArray('0b00001')
```

The general case of stripping all zeros from the start would probably be

`a = a[-a.uint.bit_length():]`

which uses the `bit_length` property of the Python `int`.

**Comment by obarak** [comment]
Thanks for the quick answer!
I still feel it makes sense to add a fill method, and I completely agree with you on keeping it simple. I think that if you add `lpad(len)` and `rpad(len)` that *always* pad with zeros you'll cover 95% of use-cases, and it seems more natural to me than initializing a new array.

**Comment by scott-griffiths** [comment]
Related to Issue #98, which was a similar idea for a fill method. I closed that one as it didn't have any traction in 11 years - maybe should have waited a couple more!

**Comment by obarak** [comment]
I doubt if I can change your mind after 11 years 😅
But *+1*-ed that issue anyway.
